cellarium/ml/core/module.py

```diff
@@ -114,17 +153,50 @@ def configure_model(self) -> None:
                 )
             )
         else:
-            transforms = None
+            transforms = tuple()
 
-        self.pipeline = CellariumPipeline(transforms)
         if model is None:
             raise ValueError(f"`model` must be an instance of {CellariumModel}. Got {model}")
-        self.pipeline.append(model)
+        self.pipeline = CellariumPipeline(cpu_transforms + transforms + (model,))  # the full pipeline
+        self.module_pipeline = self.pipeline  # training loop pipeline: initially the full pipeline
 
         if not self.hparams["is_initialized"]:
             model.reset_parameters()
             self.hparams["is_initialized"] = True
 
+        # move the cpu_transforms to the dataloader's collate_fn if the dataloader is going to apply them
+        try:
+            if hasattr(self, "trainer"):  # for some reason this check raises a RuntimeError rather than returning False
```

Comment:
We can maybe use private property `self._trainer` and judging from https://github.com/Lightning-AI/pytorch-lightning/issues/17517#issuecomment-1532325528 I think it is unlikely they will change it.

Reply:
Will do